### DISCUSSION TITLE
ESLint rule import/no-extraneous-dependencies when using @simplewebauthn/typescript-types

### DISCUSSION THREAD

**Discussion by ptmkenny** [MasterKale/SimpleWebAuthn#479] [Troubleshooting]
## Describe the issue

When using this library in my project, ESLint is identifying an issue:

`'@simplewebauthn/typescript-types' should be listed in the project's dependencies, not devDependencies.`

I am not sure if this issue is a problem with my use, ESLint, or how the types are imported.

## Reproduction Steps

Here's a very simple file to reproduce the error:

```
import * as SimpleWebauthnBrowser from '@simplewebauthn/browser';
import { RegistrationResponseJSON } from '@simplewebauthn/typescript-types';
import { WebauthnPublicKeyCredentialCreationOptions } from '../../../drupal_schema_types/webauthn-public-key-credential-creation-options.schema';

const getQueryWebauthnRegistrationResponse = async (
  regOptions: WebauthnPublicKeyCredentialCreationOptions,
): Promise<RegistrationResponseJSON> => {
  const registrationResponse =
    await SimpleWebauthnBrowser.startRegistration(regOptions);
  return registrationResponse;
};

export default getQueryWebauthnRegistrationResponse;

```

Here's the ESLint rule that triggers the error:

```
"import/no-extraneous-dependencies": ["error", {"devDependencies": false, "optionalDependencies": false, "peerDependencies": false}]
```

Specifically, `devDependencies` being false triggers the error.

## Expected behavior

I expect ESLint not to flag an error here.

The [ESLint docs](https://github.com/import-js/eslint-plugin-import/blob/v2.29.0/docs/rules/no-extraneous-dependencies.md) state:

> devDependencies: If set to false, then the rule will show an error when devDependencies are imported. Defaults to true. Type imports are ignored by default.

So I set `devDependencies` to `false` to get it to raise an error if I import dev dependencies by mistake, but since "Type imports are ignored by default", and `@simplewebauthn/typescript-types` is (I think?) a type import, then this shouldn't be triggered.  I import types for other modules elsewhere in my project and I don't have this issue.

### SimpleWebAuthn Libraries

```sh
$ npm list --depth=0 | grep @simplewebauthn
├── @simplewebauthn/browser@8.3.4
├── @simplewebauthn/typescript-types@8.3.4
```

**Comment by MasterKale** [comment]
Hey @ptmkenny, does `import type { ... } from '@simplewebauthn/typescript-types'` appease eslint?

**Reply by ptmkenny** [reply]
Yes, that resolves the issue! Thank you.
